opensearch/src/main/java/org/opensearch/sql/opensearch/request/OpenSearchRequestBuilder.java

```diff
@@ -37,111 +35,102 @@
 import org.opensearch.search.sort.SortBuilder;
 import org.opensearch.search.sort.SortBuilders;
 import org.opensearch.sql.ast.expression.Literal;
-import org.opensearch.sql.common.setting.Settings;
 import org.opensearch.sql.common.utils.StringUtils;
 import org.opensearch.sql.exception.SemanticCheckException;
 import org.opensearch.sql.expression.ReferenceExpression;
 import org.opensearch.sql.opensearch.data.type.OpenSearchDataType;
 import org.opensearch.sql.opensearch.data.value.OpenSearchExprValueFactory;
 import org.opensearch.sql.opensearch.response.agg.OpenSearchAggregationResponseParser;
-import org.opensearch.sql.planner.logical.LogicalNested;
+
 
 /**
  * OpenSearch search request builder.
  */
-@EqualsAndHashCode(callSuper = false)
+@EqualsAndHashCode
 @Getter
 @ToString
-public class OpenSearchRequestBuilder implements PushDownRequestBuilder {
+public class OpenSearchRequestBuilder implements ExecutableRequestBuilder {
 
   /**
    * Default query timeout in minutes.
    */
   public static final TimeValue DEFAULT_QUERY_TIMEOUT = TimeValue.timeValueMinutes(1L);
 
   /**
-   * {@link OpenSearchRequest.IndexName}.
+   * Search request source builder.
    */
-  private final OpenSearchRequest.IndexName indexName;
+  private final SearchSourceBuilder sourceBuilder;
 
   /**
-   * Index max result window.
+   * Query size of the request -- how many rows will be returned.
    */
-  private final Integer maxResultWindow;
+  private int requestedTotalSize;
 
   /**
-   * Search request source builder.
+   * Size of each page request to return.
    */
-  private final SearchSourceBuilder sourceBuilder;
+  private Integer pageSize = null;
```

Comment:
Use `Optional`

Reply:
I did that at first and code analysis got unhappy. Using `Integer` class is reasonable here.

From API docs:
```
Optional is primarily intended for use as a method return type where there is a clear need to represent "no result," and where using null is likely to cause errors.
```